### PR TITLE
fix(vm): apply custom timeouts on update operations

### LINKED ISSUE #1169: timeouts does not apply all the time
**Describe the bug**
I use the provider through the Pulumi wrapper where I noticed an issue. Eventually we tracked down thats its an upstream issue:
https://github.com/muhlba91/pulumi-proxmoxve/issues/266

If a timeout value set (e.g. `timeout_shutdown_vm` ) its applies for the first time, but not after. 

**To Reproduce**

1. Create a simple VM resource
2. terraform apply
3. After 180 sec, timeout occurs as it should:
```bash 
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

proxmox_virtual_environment_vm.timeouttester9000: Destroying... [id=1301]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m50s elapsed]
╷
│ Error: error waiting for VM shutdown: error retrieving task status: received an HTTP 599 response - Reason: Too many redirections
```

5. Run terraform apply one more time and its not considered again:
```bash
Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

proxmox_virtual_environment_vm.timeouttester9000: Destroying... [id=1301]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 1m50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 2m50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 3m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 3m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 3m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 3m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 3m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 3m50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 4m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 4m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 4m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 4m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 4m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 4m50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 5m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 5m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 5m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 5m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 5m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 5m50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 6m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 6m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 6m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 6m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 6m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 6m50s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 7m0s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 7m10s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 7m20s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 7m30s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 7m40s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 7m51s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 8m1s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 8m11s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 8m21s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 8m31s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 8m41s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 8m51s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 9m1s elapsed]
proxmox_virtual_environment_vm.timeouttester9000: Still destroying... [id=1301, 9m11s elapsed]
^C
Interrupt received.
Please wait for Terraform to exit or data loss may occur.
Gracefully shutting down...

Stopping operation...
╷
│ Error: error waiting for VM shutdown: error retrieving task status: failed to perform HTTP GET request (path: nodes/test-devops-proxmox02/tasks/UPID:test-devops-proxmox02:000425B4:063B204F:66017E41:qmshutdown:1301:root@pam:/status) - Reason: Get "https://10.10.119.185:8006/api2/json/nodes/test-devops-proxmox02/tasks/UPID:test-devops-proxmox02:000425B4:063B204F:66017E41:qmshutdown:1301:root@pam:/status": context canceled
│ 
│ 
╵
╷
│ Error: execution halted
│ 
│ 
╵
╷
│ Error: execution halted
│ 
```

Please also provide a minimal Terraform configuration that reproduces the issue.

```hcl

resource "proxmox_virtual_environment_vm" "timeouttester9000" {
  bios          = "seabios"
  vm_id         = "1301"
  name          = "timeouttester9000"
  scsi_hardware = "virtio-scsi-pci"
  node_name     = "test-devops-proxmox02"
  timeout_clone = 180
  timeout_create = 180
  timeout_move_disk = 180
  timeout_migrate = 180
  timeout_reboot = 180
  timeout_shutdown_vm = 180
  timeout_start_vm = 180
  timeout_stop_vm = 180

  agent {
    timeout = "3m"
  }
  cpu {
    cores   = 3
    sockets = 1
    type    = "qemu64"
  }
...

```

**Expected behavior**
Timeouts are consistent across runs.

- Single or clustered Proxmox: Clustered
- Provider version (ideally it should be the latest version): 0.50.0
- Terraform version: 1.7.5
- OS (where you run Terraform from): WSL Ubuntu 22.04
- Debug logs (`TF_LOG=DEBUG terraform apply`):

### PR DESCRIPTION
**Pull request by bpg-dev**
### What does this PR do?

Fixes VM timeout values (e.g. `timeout_shutdown_vm = 180`) being ignored on subsequent `terraform apply` runs. The first apply respects the configured timeout, but updates would fall back to the SDK's default 20-minute context timeout.

**Root cause:** The Terraform SDK wraps `UpdateContext` with `context.WithTimeout(ctx, 20*time.Minute)`, silently capping any custom timeout. A prior fix (commit `a57bd7e1`) added `context.WithoutCancel` to `vmCreate` and `vmDelete` to strip this wrapper, but `vmUpdate` was missed.

**Changes:**
1. Add `context.WithoutCancel(ctx)` to `vmUpdate` — matching the existing pattern in `vmCreate`/`vmDelete`
2. Switch timeout fields from `setDefaultIfNotSet` (uses `d.GetOk()`) to `setDefaultIfNotExists` (uses `d.GetOkExists()`) — fixes edge case where explicit `timeout = 0` was silently overwritten with defaults
3. Remove dead `setDefaultIfNotSet` function (zero callers after change 2)
4. Add acceptance test verifying timeout values persist across updates

### Contributor's Note

- [x] I have run `make lint` and fixed any issues.
- [x] I have updated documentation (FWK: schema descriptions + `make docs`; SDK: manual `/docs/` edits).
- [x] I have added / updated acceptance tests (**required** for new resources and bug fixes — see [ADR-006](docs/adr/006-testing-requirements.md)).
- [x] I have considered backward compatibility (no breaking schema changes without `!` in PR title).
- [ ] For new resources: I followed the [reference examples](docs/adr/reference-examples.md).
- [x] I have run `make example` to verify the change works (mainly for SDK / provider config changes).

### Proof of Work

Acceptance test output:

```
=== RUN   TestAccResourceVM/timeout_persistence_across_updates
=== PAUSE TestAccResourceVM/timeout_persistence_across_updates
=== CONT  TestAccResourceVM/timeout_persistence_across_updates
--- PASS: TestAccResourceVM/timeout_persistence_across_updates (3.15s)
```

Full checklist:

| Step | Status |
|------|--------|
| Build | PASSED |
| Go Lint | PASSED (0 issues) |
| Unit Tests | PASSED |
| Acceptance Tests | PASSED |
| Documentation | PASSED (no schema changes) |

### Community Note

- Please vote on this pull request by adding a :+1: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #1169

### CODE REVIEW

File: proxmoxtf/resource/vm/vm.go [hunk 1]
```diff
@@ -5476,6 +5466,9 @@ func vmUpdatePool(
 }
 
 func vmUpdate(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	// reset the default timeout for the update operation
+	ctx = context.WithoutCancel(ctx)
```
[the remark this answers is not in the document]
Comment (bpg-dev) [reply proxmoxtf/resource/vm/vm.go:5470]: /gemini, you're wrong here

1. vmUpdate already applies per-operation timeouts internally. Each operation within vmUpdate (shutdown, start, reboot, move disk, etc.) reads its own specific timeout field and creates its own context.WithTimeout. There is no code path that runs unbounded.
2. vmCreate does NOT wrap with an overall timeout either. Gemini claims this would be "consistent with vmCreate", but let's verify:

[the remark this answers is not in the document]
Comment (bpg-dev) [reply proxmoxtf/resource/vm/vm.go:5470]: Good questions. Here's the breakdown:

**Timeout coverage in `vmUpdate`:** The lifecycle operations (shutdown, start, stop, reboot, migrate) all apply their own `context.WithTimeout` from their respective timeout fields. Disk move/resize and `UpdateVM` calls use the bare context — but this is pre-existing behavior, not introduced by this PR. The same is true in `vmCreate` (see lines 2557, 2608 where `MoveVMDisk` is called with bare `ctx`). The underlying HTTP client and task waiter also have their own timeouts, so these calls won't hang indefinitely.

**`timeout_migrate`:** The comment on line 293 notes it's "essentially a timeout_update, needs to be refactored" — but currently it's only wired to the migration code path (line 5494). Repurposing it as a blanket update timeout would be a behavioral change beyond the scope of this bug fix.

Adding explicit timeouts for disk move/resize operations would be a good follow-up improvement, but it's orthogonal to this fix.

[the remark this answers is not in the document]
Comment (bpg-dev) [reply proxmoxtf/resource/vm/vm.go:5470]: /gemini does it make sense to you?
